esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1913,16 +1915,19 @@ void WsWuInfo::getWorkunitEclAgentLog(const char* fileName, const char* agentPid
                 buf.clear();
             }
 
-            //If we spot the workunit id anywhere in the tacing for this pid then assume it is the correct instance.
+            //If we spot the workunit id anywhere in the tracing for this pid then assume it is the correct instance.
             if(!wuidFound && strstr(line.str(), wuid.str()))
+            {
+                pidPtr = pPid;//remember offset of PID within line
```

Comment:
Seems to me this will only work if the buffer of line.str() is not reallocated. You could store pidOffset = pPid-line.str() and compare that.

You are also relying on all columns prior to the pid being fixed width - Is that normally true?